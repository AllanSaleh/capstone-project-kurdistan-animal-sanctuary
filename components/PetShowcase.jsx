import React from "react";
import Link from "next/link";

const PetShowcase = () => {
  const img = [
    {
      name: "Tomas",
      "image-path": "./cat1.jpeg",
      url: "https://i.imgur.com/fWhk9kd.jpeg",
    },
    {
      name: "Lucy",
      "image-path": "./cat2.jpeg",
      url: "https://i.imgur.com/uvFEcJN.jpeg",
    },
    {
      name: "Zeus",
      "image-path": "./dog.jpeg",
      url: "https://i.imgur.com/wqF89Ol.jpeg",
    },
  ];

  return (
    <section className="petshowcase bg-secondary font-Quicksand">
      <div className="px-24">
        <h1 className="text-center pt-6 pb-20 text-7xl font-bold text-primary">
          Make a Pet Happy
        </h1>
        <div className="flex flex-row justify-around">
          {img.map((item) => (
            <Link href="./">
              <div className="card relative rounded-lg cursor-pointer">
                <img
                  className=" img-showcase rounded-lg"
                  src={item["image-path"]}
                  alt="animal profile "
                />

                <div className="absolute left-16 bottom-6 bg-white w-3/5 h-12 bg-opacity-90 rounded-lg">
                  <p className="text-center opacity-100 text-4xl text-secondary">
                    {item.name}
                  </p>
                </div>
              </div>
            </Link>
          ))}
        </div>
      </div>
    </section>
  );
};
export default PetShowcase;
