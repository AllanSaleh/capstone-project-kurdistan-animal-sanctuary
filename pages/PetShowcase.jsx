import React from "react";

const PetShowcase = () => {
  const img = [
    "https://i.imgur.com/fWhk9kd.jpeg",
    "https://i.imgur.com/uvFEcJN.jpeg",
    "https://i.imgur.com/fWhk9kd.jpeg",
  ];

  return (
    <section className="petshowcase bg-secondary ">
      <div className="px-24">
        <h1 className="text-center pt-6 pb-20 text-7xl font-bold text-primary">
          {" "}
          Make a Pet Happy{" "}
        </h1>
        <div className="flex flex-row justify-around">
          {" "}
          {img.map((item) => (
            <div className="card relative rounded-lg">
              {" "}
              <a href={item}>
                {" "}
                <img
                  className=" img-showcase rounded-lg"
                  src={item}
                  alt="animal profile "
                />
              </a>
              <div className="absolute left-16 bottom-6 bg-white w-48 h-12 bg-opacity-90 rounded-lg">
                <p className="text-center opacity-100 text-4xl text-secondary">
                  {" "}
                  name{" "}
                </p>
              </div>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
};
export default PetShowcase;
