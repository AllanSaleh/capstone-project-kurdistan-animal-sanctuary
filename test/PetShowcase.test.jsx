import React from "react";
import "babel-polyfill";
import renderer from "react-test-renderer";
import { test, expect } from "@jest/globals";
import PetShowcase from "../components/PetShowcase";

test("proparly renders the PetShowcase component", () => {
  const tree = renderer.create(<PetShowcase />).toJSON();
  expect(tree).toMatchSnapshot();
});
