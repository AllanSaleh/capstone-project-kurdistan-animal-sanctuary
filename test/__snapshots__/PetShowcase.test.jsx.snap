// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proparly renders the PetShowcase component 1`] = `
<section
  className=" w-full h-5/6 pb-16 bg-secondary font-Quicksand"
>
  <div
    className=""
  >
    <h1
      className="text-center mx-auto w-full pt-6 pb-10 text-3xl w-72 font-bold text-primary md:text-7xl "
    >
      homePage.makeApetHappy
    </h1>
    <div
      className="flex flex-col justify-around items-center px-20 md:px-24 md:flex-row"
    />
  </div>
</section>
`;
